crates/turborepo-lib/src/cli/mod.rs

```diff
@@ -758,6 +761,192 @@ impl RunArgs {
             (None, None) => None,
         }
     }
+
+    pub fn track(&self, telemetry: &CommandEventBuilder) {
```

Comment:
I've looked for every feature of Structs to make this easier, but this is all I've got so far. But it is painful.

Reply:
What you are probably looking for is a [procedural macro](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) that allows you to generate code based on the `struct` definition. This is fairly advanced and probably only makes sense if we end up doing this a lot and it becomes a pain to maintain.

There might be some crates that make writing this easier, but this is an area I don't know much about.

Reply:
Something that might make this easier is to use a macro to generate these calls. e.g.
```
macro_rules! track_usage {
           ($field:ident) => {
                if self.$field {
                    telemetry.track_arg_usage(stringify!($field), true);
                }
            };
            ($field:ident, $is_used:expr) => {
                if $is_used(&self.$field) {
                    telemetry.track_arg_usage(stringify!($field), true);
                }
            };
        }
```

This allows for `track_usage!(skip_infer);` which expands to:
```
if self.skip_infer {
    telemetry.track_arg_usage("skip_infer", true);
}
```

And for providing a "is set" function for non bool types e.g. `track_usage!(api, Option::is_some);` which expands to:
```
if (Option::is_some)(&self.api) {
    telemetry.track_arg_usage("api", true);
}
```

Reply:
Done!